### DISCUSSION TITLE
Agents (Windows nor Linux) Will Not Connect To New MC Install (hosted on Windows)

### DISCUSSION THREAD

**Discussion by DLTwiStar** [Ylianst/MeshCentral#6311] [Issues] [answered]
Hello All, 

is anyone getting this issue where trying to add new agents to a new setup give this error? and why is it that the WSS address is a IPv6 Address instead or the domain name? is that normal?

`[REDACTED]@[REDACTED]:/usr/local/mesh_services/meshagent$ sudo ./meshagent 
Generating Certificate...
AutoRetry Connect in 1619 milliseconds
AutoRetry Connect in 2573 milliseconds
Connecting to: wss://[fe80::ac61:1cca:2848:127f%2]:443/agent.ashx
Mesh Server Connection Error [-1]
AutoRetry Connect in 4651 milliseconds
`

Server: Hosting on Windows Server 2022
Domain is Hosted Behind Nginx Reserve Proxy (Followed the guild and had it working before new install)

**Comment by si458** [comment]
First you need to check your config.json has a cert value set,
Second is ur meshcentral server ipv6 only or mix?
Third, If ipv6+ipv4 is ipv6 primary or secondary?
Forth, check the meshagent.msh file for the serverurl (sits next to meshagent normally)

**Reply by DLTwiStar** [reply]
Hello @si458, thank for your reply.

1. i checked the _cert value: good
2. no, im using IPv4 not IPv6.
3. IPv4 should be the primary.
4. i dont see "serverurl" in the meshagent.msh file...


```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "_cert": "[REDACTED].[REDACTED].org",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "[REDACTED]",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TlsOffload": "[REDACTED, localhost and Nginx proxy IP]"
  },
  "domains": {
    "": {
      "certUrl": "https://[REDACTED].[REDACTED].org",
      "allowedOrigin": "[REDACTED].[REDACTED].org",
      "_title": "TwiStar",
      "_title2": "Remote Management",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  }
}
``

**Reply by si458** [reply, marked as answer]
right so, im ASSUMING you are using meshcentral externally
1. you should **always** uncomment `_cert` to `cert` and set a value
2. you add `"trustedProxy": "IPOFPROXYHERE"` under `"settings"` so meshcentral can get the correct IP addresses of agents
3. you should also uncomment `"_sessionkey"` to `"sessionkey"` and set it to a value thats unique and SECRET,
this just helps with remembering sessions between restarts of meshcentral
 
then restart meshcentral and uninstall meshagent on ur remote devices, then redownload meshagent again, and reinstall and the msh file should now include a serverurl

**Comment by DLTwiStar** [comment]
BINGO! Got it working! Thank you so much @si458 and sorry for my stupidity... :/ Was up late trying to figure it out and i guess my brain already shut off for the night.

For anyone who host MC on Windows and runs it via Nginx Reverse Proxy, here is was your config should look like for MC as of version 1.1.27: 

[@si458 - please correct this if needed. you may delete this line. Thanks again!]

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "[FQDN]",
    "_WANonly": true,
    "_LANonly": true,
    "sessionKey": "[RANDOM STRING OF CHAR]",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TlsOffload": "[LOCALHOST AND PROXY IP]",
    "trustedProxy": "[IP OF PROXY]"
  },
  "domains": {
    "": {
      "certUrl": "https://[FQDN]",
      "allowedOrigin": "[FQDN]",
      "_title": "MyServer",
      "_title2": "MyServer",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  }
}
```
